tests/test_get_active_task_definition_by_tag.py

```diff
@@ -0,0 +1,164 @@
+import unittest
+from unittest.mock import Mock, patch
+
+import boto3
+
+from actions_helper.commands.get_active_task_definition_by_tag import (
+    NonSingleValueError,
+    Tag,
+    format_tags,
+    get_active_task_definition_arn_by_tag,
+)
+from tests.utils import TEST_APPLICATION_ID
+
+
+class GetTaskDefinitionByTagTestCase(unittest.TestCase):
+    @patch.object(boto3, "client")
+    def setUp(self, boto3_client):
+        self.ecs_client = boto3_client
+        self.pulumi_tag = {"key": "created_by", "value": "Pulumi"}
+
+    def test_format_tags(self):
+        with self.subTest("Invalid tag format"), self.assertRaises(expected_exception=ValueError):
+            format_tags("invalid-tag")
+            format_tags("key:value key:value")
+
+        with self.subTest("valid tag format"):
+            tags = format_tags("key:value, key2:value2")
+            self.assertEqual(len(tags), 2)
+            self.assertTupleEqual(tags, (Tag("key", "value"), Tag("key2", "value2")))
+
+    def test_get_active_definition_by_tag(self):
+        with (
+            self.subTest("Tag does not exist"),
+            self.assertRaises(NonSingleValueError),
+            patch.object(
+                self.ecs_client,
+                "list_task_definitions",
+                side_effect=Mock(return_value={"taskDefinitionArns": []}),
+            ),
+        ):
+            get_active_task_definition_arn_by_tag(
+                ecs_client=self.ecs_client,
+                task_definition_family_prefix=TEST_APPLICATION_ID,
+                task_definition_tags="created_by:Pulumi",
+                allow_initial_deployment=False,
+            )
+
+        with (
+            self.subTest("More than 1 definition with same Tag exist"),
+            patch.object(
+                self.ecs_client,
+                "list_task_definitions",
+                side_effect=Mock(return_value={"taskDefinitionArns": ["dummy", "dummy"]}),
+            ),
+            patch.object(
+                self.ecs_client,
+                "describe_task_definition",
+                side_effect=Mock(
+                    return_value={"taskDefinition": {"taskDefinitionArn": "dummy"}, "tags": [self.pulumi_tag]},
+                ),
+            ),
+            self.assertRaises(NonSingleValueError),
+        ):
+            get_active_task_definition_arn_by_tag(
+                ecs_client=self.ecs_client,
+                task_definition_family_prefix=TEST_APPLICATION_ID,
+                task_definition_tags=f"{self.pulumi_tag['key']}:{self.pulumi_tag['value']}",
+                allow_initial_deployment=False,
+            )
+
+        with (
+            self.subTest("Tag exist"),
```

Comment:
```suggestion
            self.subTest("Tag exists"),
```

Can you please also check the other subTest descriptions? We can also quickly go over them together if it is easier.